src/view/DashboardView.java

```diff
@@ -0,0 +1,202 @@
+package view;
+
+import interface_adapter.dashboard.DashboardState;
+import interface_adapter.dashboard.DashboardViewModel;
+import interface_adapter.dashboard.ProjectData;
+import interface_adapter.add_project.AddProjectController;
+
+
+import javax.swing.*;
+import java.awt.*;
+import java.awt.event.ActionEvent;
+import java.awt.event.ActionListener;
+import java.beans.PropertyChangeEvent;
+import java.beans.PropertyChangeListener;
+import java.text.ParseException;
+import java.util.ArrayList;
+import javax.swing.text.MaskFormatter;
+
+public class DashboardView extends JFrame implements PropertyChangeListener {
+    private final DashboardViewModel dashboardViewModel;
+    private JPanel dashboardPanel;
+    private ArrayList<ProjectData> projectsList;
+
+    public DashboardView(DashboardViewModel dashboardViewModel, AddProjectController addProjectController) {
+        this.dashboardViewModel = dashboardViewModel;
+
+        DashboardState dashboardState = dashboardViewModel.getState();
+
+        dashboardViewModel.addPropertyChangeListener(this);
+
+        projectsList = dashboardState.getProjects();
+
+        displayAllProjects();
+
+        setTitle("Project Dashboard");
+        setSize(600, 400);
+        setDefaultCloseOperation(JFrame.EXIT_ON_CLOSE);
+        setLayout(new BorderLayout());
+        setResizable(false);
+
+        dashboardPanel = new JPanel();
+        dashboardPanel.setLayout(new BoxLayout(dashboardPanel, BoxLayout.Y_AXIS));
+
+        JScrollPane scrollPane = new JScrollPane(dashboardPanel);
+        scrollPane.setVerticalScrollBarPolicy(ScrollPaneConstants.VERTICAL_SCROLLBAR_ALWAYS);
+
+        add(scrollPane, BorderLayout.CENTER);
+
+        updateEmptyDashboardLabel();
+
+        JButton addProjectButton = new JButton("Add Project");
+        addProjectButton.addActionListener(new ActionListener() {
+            @Override
+            public void actionPerformed(ActionEvent e) {
```

Comment:
Sounds good! Since you are going to create another pull request after this has been merged to main with your addition of toggleNotification, I'll leave this feature up to your branch.